site/content/extend/plugins/best-practices.md

```diff
@@ -75,26 +75,45 @@ For examples of custom settings see: Demo Plugin [`CustomSetting`](https://githu
 
 ## How can I review the entire code base of a plugin?
 
-Carry out the following steps:
+Sometimes, you have been working on a personal repository for a new plugin, most probably based on the [mattermost-plugin-starter-template](https://github.com/mattermost/mattermost-plugin-starter-template/) repo. As it was a personal project, you may have pushed all of your commits directly commits to `master`. And now that it's functional, you need a reviewer to take a look at the whole thing.
 
-1. Take a backup of project directory
-2. Create a dummy-master branch with no code:
+For this, it is useful to create a PR with only the commits you added. Follow these steps to do so:
+
+0. Make sure that you are in the `master` branch, and that the remote is updated.
+
+   ```
+   git checkout master
+   git push origin master
+   ```
+   
+1. Create a new branch (it will be an exact copy of `master`) and push it to the remote:
+
+   ```
+   git checkout -b review
```

Comment:
Could we simplify this and the earlier `checkout master` to just:
```
git fetch
git branch review --no-track origin/master
git push origin review
```

Reply:
Here I was assuming that the developer may have local changes not pushed to the remote. But maybe we can just do
```
git branch review --no-track master
git push origin review
```
What do you think?